Constraints reb

rebadam opened the pull request:
I added constraints handling to the AMSWorker. This entails an extra method SetConstraints. Also, the GeometryOptimization now looks for a message about the convergence. If it finds it, it just adds it to the results object.

I added this to the docs, and I made an example. To have this example work, I made a minor update to the plot_grid_molecules function (by Giullio). It now guesses bonds only if the molecule does not have bonds yet. In addition, it removes the conformer from the rdkit molecule, since this generally results in better 2D images.

I also made minor changes to a previous example, which I just included here because I did not want to make a whole separate pull request for it.

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107 Constraints and convergence info in AMSWorker

- Added a method to the amsworker to detect constraints
- Added convergence info to the results for geometry optimization

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO102: Changed name of optimizationResults

- Matches change in AMSPipe

rebadam committed:
Merge branch 'trunk' into constraintsREB

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO102: Added an example of constrained amsworker GO

- I added docs for the new Constraints feature in AMSWorker geometry optimizations
- I added an example
- I changed the plot_grid_molecules a bit:
  * Only guess bonds if the bonds are not yet there
  * Remove the conformers. This results in nicer looking 2D structures

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Reformatted

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Small fix to notebook

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Beauty changes to example

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Fixed black issue I missed

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Updated base image for plot_grid_molecules

GiulioIlBen reviewed (approved):
On my side these changes improves upon previous work. So it is great!

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Moved SetConstraints to GeometryOptimization in AMSWorker

rebadam committed:
Rosa Bulo (REB) SCMSUITE-- SO107: Constraints working in AMSWorkerPool

- Made constraints work with the AMSWorkerPool
- Added docstring in GeometryOptimization method

robertrueger reviewed (approved):
Looks good to me now! :)
